### ISSUE TITLE
permettre l'ajout de fichier h5p sur le wordpress d'osm france

### ISSUE THREAD

**Issue by vinber** [osm-fr/infrastructure#449]
dans l'objectif de donner à voir le "mooc" d'initiation à OSM  fait avec la DANE de Lyon, j'ai activé le plugin h5p de wordpress, j'ai tenté l'import de fichier h5p, cela ne fonctionne pas 
![image](https://github.com/osm-fr/infrastructure/assets/890869/15d56e10-dffe-4731-8cc2-068f6894bd2f)

cependant la création d'une activité fonctionne donc le module marche, c'est bien l'import qui ne fonctionne pas.

- [ ] taille de fichiers (ils font souvent 4 ou 5 Mo)
- [ ] type de fichier limité format .h5p
- [ ] ...

**Comment by bristow** [comment]
Quand je teste via Librairies à téléverser, il met ce message d'erreur : 

`The uploaded file exceeds the upload_max_filesize directive in php.ini`

à voir donc dans ce fichier. Merci !

**Comment by Marc-marc-marc** [comment]
valeur actuelle `/etc/php/7.4/cli/php.ini:upload_max_filesize = 2M`
je monte a 12Mo et systemctl restart php7.4-fpm.service

**Comment by vinber** [comment]
et en tentant avec un tout petit fichier h5p de 0.5Mo cela fonctionne.

donc ce n'est pas le format mais bien en lien avec la taille ...

**Comment by vinber** [comment]
magnifique exemple https://www.openstreetmap.fr/test-h5p/

**Comment by vinber** [comment]
test sur chromium, ne fonctionne pas

**Comment by vinber** [comment]
est-ce que cela peut-être testé ? comment savoir la limite coté nginx ?

**Comment by bristow** [comment]
@cquest a vérifié le nginx lors du sotm et c'est réglé à 50 Mo. Bizarre...

**Comment by jocelynj** [comment]
Effectivement, ça pourrait être une idée d'uploader par un autre canal (je préférerais ssh que ftp ici), mais est-ce qu'on saurait l'envoyer à wordpress après ?

@vinber : tu pourrais m'envoyer le fichier que je teste, ainsi que la page d'admin à utiliser ?

**Comment by bristow** [comment]
J'utilise le plugin WordPress H5P sur 2 autres sites et je n'ai aucun soucis pour remonter ces fichiers. Le SSH ou FTP peuvent être une solution mais très contraignante car je ne suis pas sûr que l'on puisse récupérer facilement l'archive ensuite. Vaudrait mieux trouver ce qui bloque...

@jocelynj : 
Voici un fichier h5p pour tester l'upload : https://nuage03.apps.education.fr/index.php/s/dZYwYxMANWoEE3k

La page à utiliser est ici : https://www.openstreetmap.fr/wp-admin/admin.php?page=h5p_new

Merci :pray:

**Comment by vinber** [comment]
merci @m2589972 de la proposition. Si effectivement on peut passer par la voie pour les "humains" c'est plus simple par la suite :) mais si tu veux jouer il y a sans doute plein de choses à faire dans les issues ...

**Comment by jocelynj** [comment]
Je pense que le problème est ailleurs, parce que le débugguer réseau de firefox montre que le fichier .h5p n'est même pas envoyé avant l'affichage de l'erreur.

**Comment by jocelynj** [comment]
Ok, c'était bien un souci sur la config serveur. J'ai changé ceci sur la VM hébergeant le wordpress:
```
--- a/php/7.4/fpm/php.ini
+++ b/php/7.4/fpm/php.ini
@@ -691,7 +691,7 @@ auto_globals_jit = On
 ; Its value may be 0 to disable the limit. It is ignored if POST data reading
 ; is disabled through enable_post_data_reading.
 ; http://php.net/post-max-size
-post_max_size = 8M
+post_max_size = 64M
 
 ; Automatically add files before PHP document.
 ; http://php.net/auto-prepend-file
@@ -843,7 +843,7 @@ file_uploads = On
 
 ; Maximum allowed size for uploaded files.
 ; http://php.net/upload-max-filesize
-upload_max_filesize = 2M
+upload_max_filesize = 64M
```

**Comment by jocelynj** [comment]
@vinber : tu pourrais retester ?

**Comment by bristow** [comment]
ça fonctionne :tada: Merci @jocelynj !!

**Comment by vinber** [comment]
@bristow mais tu t'appelles pas @vinber :) !!!

merci à tout le monde surtout
